testutils/clusterlock/main.go

```diff
@@ -60,96 +64,153 @@ type TestClusterLocker struct {
 	clientset kubernetes.Interface
 	namespace string
 	buidldId  string
+	ctx       context.Context
+}
+
+type Options struct {
+	Namespace string
+	IdPrefix string
+	Context context.Context
 }
 
-func NewTestClusterLocker(clientset kubernetes.Interface, namespace string) (*TestClusterLocker, error) {
-	if namespace == "" {
-		namespace = LockDefaultNamespace
+func NewTestClusterLocker(clientset kubernetes.Interface, options Options) (*TestClusterLocker, error) {
+
+
+	if options.Namespace == "" {
+		options.Namespace = LockDefaultNamespace
+	}
+	if options.Context == nil {
+		options.Context = context.Background()
 	}
-	_, err := clientset.CoreV1().ConfigMaps(namespace).Create(defaultConfigMap)
+	buildId := options.IdPrefix + uuid.New().String()
```

Comment:
can we correlate build id with something from the actual build?

Reply:
It needs to be a uuid, but an option for a prefix can be passed in which can be used to ID